### PR TITLE
[GH-15803] fix: migrate this.ref.dotMenu to React.createRef

### PR DESCRIPTION
**Pull request by jufab**
#### Summary

Migrate this.refs.dotMenu to React.createRef 

#### Ticket Link

Jira ticket: https://mattermost.atlassian.net/browse/MM-29314
Fixes https://github.com/mattermost/mattermost-server/issues/15803

#### Related Pull Requests

None

#### Screenshots

None

```release-note
none
```

### CODE REVIEW

**Review by sbishel — APPROVED** [review]

**Review by calebroseland — APPROVED** [review]

File: components/post_view/post_info/post_info.jsx [hunk 1]
```diff
@@ -134,6 +134,7 @@ export default class PostInfo extends React.PureComponent {
         };
 
         this.postHeaderRef = React.createRef();
+        this.dotMenu = React.createRef();
```
Comment (Willyfrog) [inline components/post_view/post_info/post_info.jsx:137]: In order to be conssistent with other PRs can you rename it like
```suggestion
        this.dotMenuRef = React.createRef();
```

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Hello! and thanks for contributing, I've added a couple of changes that we need in order to merge the PR, minor things but they are needed :)

- [ ] in line https://github.com/mattermost/mattermost-webapp/pull/6667/files#diff-7379f4efc36537be34c4c4f1e8a2ddda465884311178b290fac653e46535a47eR244
you forgot to change the string reference for the new generated reference.

- [ ] Happens the same thing in https://github.com/mattermost/mattermost-webapp/pull/6667/files#diff-4b4a9daf51df4d5dc33e829184b91c5bb43805378bb7b1db4f3caa7f3702db60R346

- [ ] Also at the beggning and end of both files, there is a comment to enable/disable the rule that looks like:
```js
/* eslint-disable react/no-string-refs */
```

can you remove those comments?

File: components/post_view/post_info/post_info.jsx (hunk 1 above)
Comment (jufab) [reply components/post_view/post_info/post_info.jsx]: Ok!

**Comment by jufab** [comment]
No Problem.
It's done in this commit.

**Review by Willyfrog — APPROVED** [review]
Looking good! thanks for contributing :)

**Comment by prapti** [comment]
In test queue.

**Comment by jfrerich** [comment]
Thanks for your contribution, @jufab! QA has been busy with hacktoberfest, but will get to your PR soon!

**Comment by Willyfrog** [comment]
hi @jufab sorry it is taking so long, I'm afraid there is a ton of contributions in hacktoberfest and it takes longer than usual. Unfortunately this also meant that we now have a conflict in your contribution, can you fix it so the process can continue?

**Comment by jufab** [comment]
@Willyfrog No problem, I understand!
It's done 😉

**Comment by stylianosrigas** [comment]
Removing setup cloud server label for maintenance purposes. Feel free to reapply if needed :)

**Review by stevemudie — APPROVED** [review]
Tested and passed.

Able to access and use items in the ⋮ menu on posts in center and RHS.

**Comment by angeloskyratzakos** [comment]
Hey all, I will remove the tag `Setup Cloud Test Server` for maintenance reasons. Feel free to re-add it when you need it. Thank you

**Comment by jasonblais** [comment]
@Willyfrog are you okay if I assign to you to help with merging?

**Comment by Willyfrog** [comment]
seems like auto-merge is not working :/

**Pull request merged**